### PR TITLE
added check for empty arrays when resolving queries

### PR DESCRIPTION
**Pull request by danrot**
This PR adds the check for empty arrays as suggested in https://github.com/jackalope/jackalope/issues/322#issuecomment-268827882

### CODE REVIEW

**Comment by danrot** [comment]
@dbu There are no tests for this class, are there?

File: src/Jackalope/Query/Row.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ public function getValue($columnName)
         // single-valued properties. We join the values when it's a string
         // for multi-values boolean/binary values we can't provide a
         // defined result so we return null
-        if (is_array($value)) {
+        if (is_array($value) && count($value)) {
             if (is_scalar($value[0]) && !is_bool($value[0])) {
```
Comment (dbu) [inline src/Jackalope/Query/Row.php]: the count should go as first check on this line, not in the outer if. we want to return null.

**Comment by dbu** [comment]
there are no unit tests. we focussed on the functional phpcr-api-tests. those test a lot of this, though maybe not every edge case.

File: src/Jackalope/Query/Row.php (hunk 1 above)
Comment (danrot) [inline src/Jackalope/Query/Row.php]: I am not sure if I understand you correctly. Should it be a separate if, or should I just swap the operands?

Comment (dbu) [inline src/Jackalope/Query/Row.php]: sorry. i meant to move count() into this if, so it becomes `if count($value) && is_scalar($value[0]) && !is_bool($value[0])) {`.

we should return null for an empty array.

**Comment by danrot** [comment]
@dbu Is it now ok?

**Comment by dbu** [comment]
looks good, thanks a lot!

**Comment by danrot** [comment]
Thank you for merging :-)
